pkg/services/datasources/service/store.go

```diff
@@ -162,12 +161,6 @@ func (ss *SqlStore) DeleteDataSource(ctx context.Context, cmd *datasources.Delet
 			}
 
 			cmd.DeletedDatasourcesCount, _ = result.RowsAffected()
-
-			// Remove associated AccessControl permissions
-			if _, errDeletingPerms := sess.Exec("DELETE FROM permission WHERE scope=?",
-				ac.Scope(datasources.ScopeProvider.GetResourceScope(ds.UID))); errDeletingPerms != nil {
-				return errDeletingPerms
-			}
```

Comment:
Note: removing this, as it's not safe - if data sources with the same UID exists in multiple orgs, we'd be deleting the permissions for all of these data sources. `DeleteDataSource` from DS store is only called from DS service which also calls access control function to clean up data source permissions, so this is safe.